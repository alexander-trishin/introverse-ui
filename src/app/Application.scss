body,
body > div {
    height: 100%;
    width: 100%;
    margin: 0;
    padding: 0;
    background-color: #19181d;
}

.circle {
    height: 400px;
    width: 400px;
    border-radius: 50%;
    background: linear-gradient(#313247 0%, #19181d 30%);
    position: absolute;
    top: 20%;
    left: 35%;
}

.circle::before,
.circle::after {
    content: '';
    height: 400px;
    width: 400px;
    border-radius: 50%;
    background: linear-gradient(#ffd1da 0%, #ff849d 5%, #2d2133 25%);
    position: absolute;
    top: 42%;
    left: 50%;
    transform: translate(-50%, -50%);
    filter: blur(7px);
    z-index: -1;
}

.circle:after {
    width: 415px;
    top: 35%;
    filter: blur(14px);
    opacity: 0.3;
}

.message {
    font-family: sans-serif;
    font-size: 20px;
    color: #fff;
    letter-spacing: 45px;
    position: absolute;
    top: 50%;
    left: 50%;
    text-align: center;
    transform: translate(-50%);
}
