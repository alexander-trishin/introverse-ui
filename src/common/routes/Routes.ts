const Routes = {
    Error: {
        NotFound: '/error/not-found',
        SomethingWentWrong: '/error/something-went-wrong'
    },
    Page: {
        Home: '/'
    }
};

export default Routes;
