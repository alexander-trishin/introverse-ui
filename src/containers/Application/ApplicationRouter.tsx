import React from 'react';
import { Switch, Route } from 'react-router-dom';

import BannerPage from 'pages/BannerPage';
import HomePage from 'pages/HomePage';
import NotFoundPage from 'pages/NotFoundPage';

const ApplicationRouter = () => (
    <>
        <Switch>
            <Route exact path="/" component={HomePage} />
            <Route component={NotFoundPage} />
        </Switch>
        <Route component={BannerPage} />
    </>
);

export default ApplicationRouter;
