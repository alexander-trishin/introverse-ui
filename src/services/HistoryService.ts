import { History, createBrowserHistory } from 'history';

class HistoryService {
    private static history?: History;

    static get instance() {
        if (!this.history) {
            this.history = createBrowserHistory({
                basename: process.env.PUBLIC_URL ?? '/'
            });
        }
        return this.history;
    }
}

export default HistoryService;
