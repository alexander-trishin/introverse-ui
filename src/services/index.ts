export { default as HistoryService } from './HistoryService';
