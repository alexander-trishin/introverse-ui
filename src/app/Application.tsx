import './Application.scss';

const Application = () => {
    return (
        <>
            <div className="circle"></div>
            <div className="message">COMING SOON</div>
        </>
    );
};

export default Application;
