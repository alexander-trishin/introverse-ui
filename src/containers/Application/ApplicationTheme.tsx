import React from 'react';
import { ThemeProvider } from '@material-ui/styles';
import { createMuiTheme, responsiveFontSizes } from '@material-ui/core';

import ApplicationRouter from './ApplicationRouter';

const defaultTheme = responsiveFontSizes(createMuiTheme());

const ApplicationTheme = () => (
    <ThemeProvider theme={defaultTheme}>
        <ApplicationRouter />
    </ThemeProvider>
);

export default ApplicationTheme;
