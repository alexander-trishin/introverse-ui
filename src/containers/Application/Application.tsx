import React from 'react';
import { Helmet } from 'react-helmet';
import { CssBaseline } from '@material-ui/core';

import ApplicationTheme from './ApplicationTheme';

const Application = () => (
    <>
        <Helmet>
            <title>Introverse</title>
            <link
                rel="stylesheet"
                href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
            />
            <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />
        </Helmet>
        <CssBaseline />
        <ApplicationTheme />
    </>
);

export default Application;
