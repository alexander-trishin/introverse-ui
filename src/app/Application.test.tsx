import { render, waitFor } from '@testing-library/react';
import { History, createMemoryHistory } from 'history';

import { Routes } from 'common/routes';
import { HistoryService } from 'services';

import Application from './Application';

jest.mock('pages/MainPage/MainPage', () => () => <div data-testid="main" />);

jest.mock('pages/NotFoundPage/NotFoundPage', () => () => {
    throw new Error('test');
});

describe('<Application />', () => {
    let history: jest.SpyInstance<History>;

    beforeEach(() => {
        history = jest.spyOn(HistoryService, 'instance', 'get');
    });

    it('should render suspense fallback', () => {
        const { getByText } = render(<Application />);

        expect(getByText(/loading/i)).toBeInTheDocument();
    });

    it('should render home page', async () => {
        const historyMock = createMemoryHistory({
            initialEntries: [Routes.Page.Home]
        });

        history.mockReturnValue(historyMock);

        const { getByTestId } = render(<Application />);

        await waitFor(() => {
            expect(getByTestId('main')).toBeInTheDocument();
        });
    });

    it('should redirect to crush page in case of unhandled error', async () => {
        const historyMock = createMemoryHistory({
            initialEntries: [Routes.Error.NotFound]
        });

        history.mockReturnValue(historyMock);

        render(<Application />);

        await waitFor(() => {
            expect(historyMock.location.pathname).toBe(Routes.Error.SomethingWentWrong);
        });
    });
});
