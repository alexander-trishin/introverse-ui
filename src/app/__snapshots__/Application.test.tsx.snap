// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Application /> should match snapshot 1`] = `
<div>
  <div
    class="circle"
  />
  <div
    class="message"
  >
    COMING SOON
  </div>
</div>
`;
